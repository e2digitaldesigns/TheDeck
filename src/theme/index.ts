export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
