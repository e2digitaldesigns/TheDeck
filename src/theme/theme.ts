export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
