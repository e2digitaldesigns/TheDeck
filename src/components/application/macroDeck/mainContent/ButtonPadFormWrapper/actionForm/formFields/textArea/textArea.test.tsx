import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import _upperFirst from "lodash/upperFirst";
import FormFieldTextArea, { IntFormFieldTextAreaProps } from "./textArea";

import { ThemeProvider } from "styled-components";
import { theme } from "../../../../../../../../theme/theme";

const mocked = {
  name: "text",
  onChange: jest.fn(),
  state: {
    profileId: "profileId",
    pageId: "pageId",
    buttonPadId: "buttonPadId",
    _id: "_id",
    order: 2,
    action: "twitter",
    subAction: "",
    seconds: 0,
    url: "",
    text: "send this tweet",
    scene: "",
    layer: "",
    page: "",
    path: "",
    profile: ""
  }
};

const defaultProps = {
  name: mocked.name,
  onChange: mocked.onChange,
  state: mocked.state
};

const testSetup = (props: IntFormFieldTextAreaProps) => {
  return render(
    <ThemeProvider theme={theme}>
      <FormFieldTextArea
        name={props.name}
        onChange={props.onChange}
        state={props.state}
      />
    </ThemeProvider>
  );
};

let wrapper: any = null;
beforeEach(() => {
  wrapper = testSetup(defaultProps);
});

afterEach(() => {
  wrapper = null;
  jest.clearAllMocks();
});

describe("<FormFieldSelect/>", () => {
  it("Should render without errors", () => {
    const component = testSetup(defaultProps);
    expect(component).toMatchSnapshot();
  });

  it("Should have default values", () => {
    const label = wrapper.getByTestId("form_field_textArea__label");
    expect(label).toBeTruthy();
    expect(label.innerHTML).toBe(_upperFirst(mocked.name) + ":");

    const input = wrapper.getByTestId("form_field_textArea__input");
    expect(input).toBeTruthy();
    expect(input.value).toBe(mocked.state.text);
  });

  it("Should call onChange", () => {
    const input = wrapper.getByTestId("form_field_textArea__input");
    expect(input).toBeTruthy();
    fireEvent.change(input, { target: { value: "chill" } });
    expect(mocked.onChange).toHaveBeenCalled();
  });
});
