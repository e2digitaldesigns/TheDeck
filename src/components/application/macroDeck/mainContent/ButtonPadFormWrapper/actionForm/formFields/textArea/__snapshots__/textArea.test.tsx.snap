// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFieldSelect/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <label
          data-testid="form_field_textArea__label"
          for="text"
        >
          Text
          :
        </label>
        <textarea
          data-testid="form_field_textArea__input"
          name="text"
        >
          send this tweet
        </textarea>
      </div>
    </div>
    <div>
      <div>
        <label
          data-testid="form_field_textArea__label"
          for="text"
        >
          Text
          :
        </label>
        <textarea
          data-testid="form_field_textArea__input"
          name="text"
        >
          send this tweet
        </textarea>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <label
        data-testid="form_field_textArea__label"
        for="text"
      >
        Text
        :
      </label>
      <textarea
        data-testid="form_field_textArea__input"
        name="text"
      >
        send this tweet
      </textarea>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
